dwh/bq_to_es_projector/bq_to_es_projector.py

```diff
@@ -127,32 +144,18 @@ def transform_row(row: Dict[str, Any]) -> Tuple[str, Dict[str, Any]]:
     Ensures numerics are numeric and nested arrays are cleaned.
     """
     doc: Dict[str, Any] = {}
-    if ES_INDEX == "contrast-summary":
-        key = "contrast_id"
-        index_name = "contrast"
-    elif ES_INDEX == "dataset-summary":
-        key = "dataset_id"
-        index_name = "dataset"
-    elif ES_INDEX == "target-summary" or ES_INDEX == "target-summary-v2":
-        key = "perturbed_target_symbol"
-        index_name = "target"
-    elif ES_INDEX == "gene-summary":
-        index_name = "gene"
-        key = "gene"
-    elif ES_INDEX == "landing-page-summary":
-        index_name = "landing-page"
-        key = "summary"
-    else:
-        raise ValueError(f"Unknown index name: {ES_INDEX}")
-    if key != "summary":
-        symbol = row.get(key)
+
+    index_prefix, key_field = get_index_metadata(ES_INDEX)
+
+    if key_field != "summary":
+        symbol = row.get(key_field)
         if not symbol:
-            raise ValueError(f"Row missing '{key}'")
+            raise ValueError(f"Row missing '{key_field}'")
     else:
         symbol = "summary"
 
     numeric_int_fields, numeric_float_fields, nested_fields = get_typed_fields(
```

Comment:
nested_fields is not used, consider [PR](https://github.com/EMBL-EBI-ABC/PerturbationCatalogue/pull/279) where it's used

Reply:
I think, let's merge this one and then I'll fix all merging conflicts afterwards